zarr/storage.py

```diff
@@ -1163,6 +1163,7 @@ def listdir(self, path=None):
                             for file_name in self.fs.find(entry_path):
                                 file_path = os.path.join(dir_path, file_name)
                                 rel_path = file_path.split(root_path)[1]
+                                rel_path = rel_path.lstrip(os.path.sep)
```

Comment:
Why are we using `os.path` here, when this logic should be identical across OSs - it's explicitly "/" that we're worried about.  Is this block touched by the S3 tests?

Reply:
I had originally just put "/", but ended up following the style of the line right below this one. Perhaps we should just change both to "/"? I have to head out now, but can take a look again later today

Reply:
I would have thought "/"s, yes.